src/engine/renderer/tr_image_ktx.cpp

```diff
@@ -755,84 +765,84 @@ void SaveImageKTX( const char *path, image_t *img )
 	}
 
 	hdr.numberOfMipmapLevels = 1;
-	int i;
+	uint32_t mipWidth, mipHeight, mipDepth;
+	int mipFilter;
 	glGetTexParameteriv( target, GL_TEXTURE_MIN_FILTER,
-			     (GLint *)&i );
-	if( i == GL_NEAREST_MIPMAP_NEAREST || i == GL_NEAREST_MIPMAP_LINEAR ||
-	    i == GL_LINEAR_MIPMAP_NEAREST || i == GL_LINEAR_MIPMAP_LINEAR ) {
-
+			     (GLint *)&mipFilter );
+	if( mipFilter == GL_NEAREST_MIPMAP_NEAREST || mipFilter == GL_NEAREST_MIPMAP_LINEAR ||
+		mipFilter == GL_LINEAR_MIPMAP_NEAREST || mipFilter == GL_LINEAR_MIPMAP_LINEAR ) {
 		mipWidth = std::max(hdr.pixelWidth, 1u);
 		mipHeight = std::max(hdr.pixelHeight, 1u);
 		mipDepth = std::max(hdr.pixelDepth, 1u);
 
 		while( mipWidth > 1 || mipHeight > 1 || mipDepth > 1 ) {
 			hdr.numberOfMipmapLevels++;
 
-			if( mipWidth  > 1 ) mipWidth  >>= 1;
-			if( mipHeight > 1 ) mipHeight >>= 1;
-			if( mipDepth  > 1 ) mipDepth  >>= 1;
+			if( mipWidth  > 1 ) mipWidth  >>= 1u;
+			if( mipHeight > 1 ) mipHeight >>= 1u;
+			if( mipDepth  > 1 ) mipDepth  >>= 1u;
 		}
 	}
 
 	hdr.bytesOfKeyValueData = 0;
 
-	size = 0;
+	uint32_t size = 0;
+	int      mipSize;
 	mipWidth = std::max(hdr.pixelWidth, 1u);
 	mipHeight = std::max(hdr.pixelHeight, 1u);
 	mipDepth = std::max(hdr.pixelDepth, 1u);
-	for(unsigned i = size = 0; i < hdr.numberOfMipmapLevels; i++ ) {
-		size += 4;
+	for(uint32_t i = size = 0; i < hdr.numberOfMipmapLevels; i++ ) {
+		size += sizeof(uint32_t);
 		if( !hdr.glFormat ) {
 			glGetTexLevelParameteriv( target, i,
 						  GL_TEXTURE_COMPRESSED_IMAGE_SIZE,
 						  &mipSize );
 		} else {
 			mipSize = mipWidth * hdr.glTypeSize * components;
-			mipSize = PAD( mipSize, 4 );
+			mipSize = PAD( mipSize, 4u );
 			mipSize *= mipHeight * mipDepth;
 		}
-		size += hdr.numberOfFaces * PAD( mipSize, 4 );
+		size += hdr.numberOfFaces * PAD( mipSize, 4u );
 
-		if( mipWidth  > 1 ) mipWidth  >>= 1;
-		if( mipHeight > 1 ) mipHeight >>= 1;
-		if( mipDepth  > 1 ) mipDepth  >>= 1;
+		if( mipWidth  > 1 ) mipWidth  >>= 1u;
+		if( mipHeight > 1 ) mipHeight >>= 1u;
```

Comment:
why does the shift amount need to be unsigned?

Reply:
Decided to be consistent with `mipWidth`, `mipHeight` types, which are `uint32_t (uint32_t >>= uint32_t)`